[Cai Runxin] iP

Rachelcoll opened the pull request:
# Topaz

> “Per aspera ad astra.” ([checkout](https://sr.mihoyo.com/))

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😋 

And it is **FREE**!

Features:

 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)
If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Rachelcoll committed:
Add rename, greet and exit to chatbot

Rachelcoll committed:
Add echo to chatbot

Rachelcoll committed:
Allow add and list todos

Rachelcoll committed:
Add mark as done feature

Rachelcoll committed:
Update implementation of setUndo and setDone in Task to follow given solution

Rachelcoll committed:
Add todos, deadlines, events

Rachelcoll committed:
Add testing of text UI

Rachelcoll committed:
Add exception handling

Rachelcoll committed:
Add delete task feature

Rachelcoll committed:
Add save function after each operation

Rachelcoll committed:
Merge branch 'branch-Level-7'

Rachelcoll committed:
Add enum to represent different type of tasks

Rachelcoll committed:
Merge branch 'branch-Level-7'

Rachelcoll committed:
Update usage of enums and change time format of task

Rachelcoll committed:
Refactor using more OOP

Rachelcoll committed:
Organize code into packages

Rachelcoll committed:
Merge branch 'add-gradle-support'

Rachelcoll committed:
Add gradle to project

Rachelcoll committed:
Configure checkstyle for the project

Rachelcoll committed:
Add Junit testcases and fix error of failure to identify finish tasks in storage.load

Rachelcoll committed:
Create JAR file for Topaz

Rachelcoll committed:
Tweak the code to comply with a coding standard

Rachelcoll committed:
Add javadoc for public methods and classes

Rachelcoll committed:
Update code to meet standard in Ui.java

Rachelcoll committed:
Add feature to give users a way to find a task by searching for a keyword.

Rachelcoll committed:
Merge branch-A-JavaDoc and branch-A-StardardCoding to master and solve merge conflict

Rachelcoll committed:
Merge branch-Level-9 to master and add solve merge conflict

Rachelcoll committed:
Update method name to follow standard

Rachelcoll committed:
Update lines with more than 120 characters

vinceg4 commented on src/main/java/topaz/task/Task.java:
```diff
@@ -0,0 +1,78 @@
+ * Represents a task with a description and a completion status.
+ * A {@link Task} object has a description and a boolean flag indicating whether the task is done.
+ */
+
```
Should the blank line be removed to follow the coding standard?

Rachelcoll replied:
Already removed, thanks!

vinceg4 commented on src/main/java/topaz/ui/Ui.java:
```diff
@@ -0,0 +1,150 @@
+
+    public void showTimeParseException(DateTimeParseException dateTimeParseException) {
+        System.out.println("Invalid input date time: " + dateTimeParseException);
+    }
```
Should there be a blank line between different methods to be consistent with the other parts of the code?

vinceg4 commented on src/main/java/topaz/task/Event.java:
```diff
@@ -0,0 +1,71 @@
+    }
+
+    @Override
+    public String toFileRecord() {
```
I like how you formatted the tasks to be saved in the file 👍🏻

vinceg4 commented on src/main/java/topaz/main/TaskList.java:
```diff
@@ -0,0 +1,140 @@
+     * @param filter
+     * @return task conatins the filter keyword in description
+     */
+    public TaskList find(String filter) {
```
I like how you implemented your `find` function, I think it is simple and easy to understand 👍🏻

vinceg4 reviewed (commented):
LGTM, just some minor nits to fix.

TerenceWongsExtravaganza commented on src/main/java/topaz/main/Parser.java:
```diff
@@ -0,0 +1,60 @@
+     */
+    public static Command parse(String prompt) throws InvalidCommandException {
+
+        if (prompt.equalsIgnoreCase("help")) {
```
is there a better way to format this?

TerenceWongsExtravaganza commented on src/main/java/topaz/ui/Ui.java:
```diff
@@ -0,0 +1,150 @@
+        scanner.close();
+    }
+
+    public void showHelp() {
```
really nice addition!

TerenceWongsExtravaganza commented on src/test/java/topaz/command/CreateCommandTest.java:
```diff
@@ -0,0 +1,67 @@
+    }
+
+    @Test
+    void testCreateEvent() {
```
good amount of test cases

Rachelcoll committed:
Add initial implementation of JavaFX

Rachelcoll committed:
Add GUI to the chatbot

Rachelcoll committed:
Add GUI to the chatbot

Rachelcoll committed:
Solve merge conflict after pull from master

TerenceWongsExtravaganza reviewed (commented):
Looks good!

Rachelcoll committed:
Add improvements to code quality

Regarding coding standard, there are some return ui string with typos.
Some variables don't follow naming conventions. Also, some methods
contains unsafe practices.

Refactoring all those code improves the code quality.

The commit refactor several variables and correct typos, and add guard
clauses to several methods.

Rachelcoll committed:
Add Assertions to taskList related methods

Sometimes we may want to check if the method that calls taskList method
successfully manipulates the list inside the taskList.

In some cases programmer may forget to add new Task to the list when
writing methods for a new type of class.

Adding assertions to these methods can
verify assumptions about Internal
Invariants, Control-Flow Invariants, Preconditions, Post conditions, and
Class Invariants.

Assertions will remind the possible flaw in the terminal if it's asserted
to be false.

Rachelcoll committed:
Merge pull request #1 from Rachelcoll/branch-A-Assertions

Add Assertions to taskList related methods

Rachelcoll committed:
Merge with master and solve merge conflicts

Rachelcoll committed:
Merge pull request #2 from Rachelcoll/branch-A-CodeQuality

Improve Code Quality

Rachelcoll committed:
Set up CI

Rachelcoll committed:
Exclude check of checkstyle in gradle.yml

Rachelcoll committed:
Add C-Statistics extension

Rachelcoll committed:
Merge pull request #3 from Rachelcoll/branch-C-Statistics

Branch c statistics

Rachelcoll committed:
Improve GUI with color, picture and size

Rachelcoll committed:
Add personality to chatbot

Rachelcoll committed:
Add more function JavaDoc

Rachelcoll committed:
Add a breif user guide

Rachelcoll committed:
Update user guide and tweak GUI

Rachelcoll committed:
Update user guide and release the product
